File: Q05/main.cpp
#include<iostream>
using namespace std;

int main(){

    int n;
    cin>>n;
    int arr[n];

    for(int i=0;i<n;i++){
        cin>>arr[i];
    }

    int i=0;
    for(int j=0;j<n;j++){
        if(arr[j]!=0){
            int temp=arr[j];
            arr[j]=arr[i];
            arr[i]=temp;
            i++;
        }
    }

    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }

    return 0;
}

File: Q05/main.py
n = int(input())
arr=[]
for i in range(n):
    arr.append(int(input()))

i=0
for j in range(n):
    if arr[j]!=0:
        temp = arr[j]
        arr[j] = arr[i]
        arr[i] = temp
        i+=1
for k in range(n):
    print(arr[k],end=" ")